File: sdklibrary/src/main/java/com/loner/android/sdk/webservice/network/helper/BaseJSONParser.java
package com.loner.android.sdk.webservice.network.helper;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * This class is used as base class for other JSON Parser classes.
 */

public class BaseJSONParser {

    /**
     * Set an Empty String if the object is null.
     */
    protected String optString(final JSONObject json, final String key) {
        return json.isNull(key) ? "" : json.optString(key);
    }

    protected String optInt(final JSONObject json, final String key) {
        return json.isNull(key) ? "" : ""+json.optInt(key);
    }

    protected String optBoolean(final JSONObject json, final String key) {
        return json.isNull(key) ? "" : ""+json.optBoolean(key);
    }

    protected JSONArray optJsonArray(final JSONObject json, final String key) {
        JSONArray jsonArray =  new JSONArray();
        try {
            return json.isNull(key) ? jsonArray : json.getJSONArray(key);
        } catch (JSONException e) {
            e.printStackTrace();
        }
      return null;
    }

    protected JSONObject optJSONObject(final JSONObject json, final String key) {
        return json.isNull(key) ? null : json.optJSONObject(key);
    }
}

File: LonerSdkLibrary/src/main/java/com/loner/android/sdk/webservice/network/helper/BaseJSONParser.kt
package com.loner.android.sdk.webservice.network.helper

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

/**
 * This class is used as base class for other JSON Parser classes.
 */

open class BaseJSONParser {

    /**
     * Set an Empty String if the object is null.
     */
    protected fun optString(json: JSONObject, key: String): String {
        return if (json.isNull(key)) "" else json.optString(key)
    }

    protected fun optInt(json: JSONObject, key: String): String {
        return if (json.isNull(key)) "" else "" + json.optInt(key)
    }

    protected fun optBoolean(json: JSONObject, key: String): String {
        return if (json.isNull(key)) "" else "" + json.optBoolean(key)
    }

    protected fun optJsonArray(json: JSONObject, key: String): JSONArray? {
        val jsonArray = JSONArray()
        try {
            return if (json.isNull(key)) jsonArray else json.getJSONArray(key)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return null
    }

    protected fun optJSONObject(json: JSONObject, key: String): JSONObject? {
        return if (json.isNull(key)) null else json.optJSONObject(key)
    }
}